llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -3012,6 +3012,32 @@ struct DecomposedSelect {
 };
 } // namespace
 
+/// Folds patterns like:
+///   select c2 (select c1 a b) (select c1 b a)
+/// into:
+///   select (xor c1 c2) b a
+static Instruction *
+foldSelectOfSymmetricSelect(SelectInst &OuterSelVal,
+                            InstCombiner::BuilderTy &Builder) {
+
+  DecomposedSelect OuterSel, InnerSel;
```

Comment:
`Value *OuterCond, *InnerCond, *InnerTrueVal, *InnerFalseVal` would be simpler :)